src/Core/src/Platform/SoftInputExtensions.cs

```diff
@@ -0,0 +1,156 @@
+#endif
+
+namespace Microsoft.Maui.Platform;
+
+/// <summary>
+/// Extension methods for interacting with a platforms Soft Input Pane
+/// </summary>
+public static partial class SoftInputExtensions
+{
+	/// <summary>
+	/// If a soft input device is currently showing, this will attempt to hide it.
+	/// </summary>
+	/// <param name="targetView"></param>
+	/// <param name="token">Cancellation token</param>
+	/// <returns>
+	/// Returns <c>true</c> if the platform was able to hide the soft input device.</returns>
+	public static Task<bool> HideSoftInputAsync(this ITextInput targetView, CancellationToken token)
+	{
+#if NETSTANDARD
+		return Task.FromResult(false);
+#else
+		token.ThrowIfCancellationRequested();
+		if (!targetView.TryGetPlatformView(out var platformView, out _, out _))
+		{
+			return Task.FromResult(false);
+		}
+
+		return Task.FromResult(HideSoftInput(platformView));
+#endif
+	}
+
+	/// <summary>
+	/// If a soft input device is currently hiding, this will attempt to show it.
+	/// </summary>
+	/// <param name="targetView"></param>
+	/// <param name="token">Cancellation token</param>
+	/// <returns>
+	/// Returns <c>true</c> if the platform was able to show the soft input device.</returns>
+	public static Task<bool> ShowSoftInputAsync(this ITextInput targetView, CancellationToken token)
+	{
+#if NETSTANDARD
+		return Task.FromResult(false);
+#else
+		token.ThrowIfCancellationRequested();
+
+		if (!targetView.TryGetPlatformView(out var platformView, out var handler, out var view))
+		{
+			return Task.FromResult(false);
+		}
+
+		if (!view.IsFocused)
+		{
+			var showKeyboardTCS = new TaskCompletionSource<bool>();
+
+#pragma warning disable CS0618
+			handler.Invoke(nameof(IView.Focus), new FocusRequest(false));
+#pragma warning restore CS0618
+
+			handler.GetRequiredService<IDispatcher>().Dispatch(() =>
+			{
+				try
+				{
+					var result = platformView.ShowSoftInput();
+					showKeyboardTCS.SetResult(result);
+				}
+				catch (Exception e)
+				{
+					showKeyboardTCS.SetException(e);
+				}
+			});
+
+			return showKeyboardTCS.Task.WaitAsync(token);
+		}
+		else
+		{
+			var result = platformView.ShowSoftInput();
+			return Task.FromResult(result).WaitAsync(token);
+		}
+#endif
+	}
+
+	/// <summary>
+	/// Checks to see if the platform is currently showing the soft input pane
+	/// </summary>
+	/// <param name="targetView"></param>
+	/// <returns>
+	/// Returns <c>true</c> if the soft input device is currently showing.</returns>
+	public static bool IsSoftInputShowing(this ITextInput targetView)
+	{
+		if (!targetView.TryGetPlatformView(out PlatformView? platformView, out _, out _))
+		{
+			return false;
+		}
+
+		return platformView.IsSoftInputShowing();
+	}
+
+	static bool TryGetPlatformView(this ITextInput textInput,
+									[NotNullWhen(true)] out PlatformView? platformView,
+									[NotNullWhen(true)] out IPlatformViewHandler? handler,
+									[NotNullWhen(true)] out IView? view)
```

Comment:
nitpick: indentation

Reply:
https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui.Core/Platform/KeyboardExtensions/KeyboardExtensions.shared.cs#L102-L106

This is the same indentation as the original code